test/integration/agent/cniserver_test.go

```diff
@@ -660,9 +685,24 @@ func TestAntreaServerFunc(t *testing.T) {
 			addresses:       []string{"10.1.2.100/24,10.1.2.1,4"},
 			Routes:          []string{"10.0.0.0/8,10.1.2.1", "0.0.0.0/0,10.1.2.1"},
 		},
+		{
+			name:       "Prometheus metrics test with ADD/DEL/CHECK for 0.4.0 config",
+			CNIVersion: "0.4.0",
+			// IPv4 only
+			ranges: []rangeInfo{{
+				subnet: "10.1.2.0/24",
+			}},
+			expGatewayCIDRs: []string{"10.1.2.1/24"},
+			addresses:       []string{"10.1.2.100/24,10.1.2.1,4"},
+			Routes:          []string{"10.0.0.0/8,10.1.2.1", "0.0.0.0/0,10.1.2.1"},
+		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			if strings.Contains(tc.name, "Prometheus") {
```

Comment:
why don't we add a new `validateMetrics` boolean field to the `testCase` struct?

Reply:
Just went with slightly hacky approach, since it is only one testcase. Added a boolean in testcase.